src/backings/structural/abstract.ts

```diff
@@ -71,4 +71,10 @@ export class StructuralHandler<T extends object> {
   toJson(value: T, options?: IJsonOptions): Json {
     throw new Error("Not implemented");
   }
+  maxSize(): number {
```

Comment:
can you move these methods to the type objects, eg: move from `src/backings/structural` to `src/types/composite` and `src/types/basic`? These methods would be declared right next to the `size` method for each type

Reply:
I moved minSize()/maxSize() next to size(). Everything started from `src/type` but for composite, it delegates its implementation to structural backing, just like `size()`